src/main/java/scribbles/Scribbles.java

```diff
@@ -0,0 +1,227 @@
+                ui.taskMissingInformationMessage();
+            } catch (FileNotFoundException e) {
+                ui.fileNotFoundMessage();
+            }
+            break;
+        case "deadline":
+            if (parsedInput.isMissingDeadlineInformation()) {
+                ui.taskMissingInformationMessage();
+            } else {
+                try {
+                    String description = parsedInput.getDeadlineDescription();
+                    LocalDateTime by = parsedInput.getDeadlineBy();
+                    taskList.addTask(new Deadline(description, false, by));
+                    ui.confirmTaskAddition(taskList);
+                    storage.saveFileData(taskList);
+                } catch (IndexOutOfBoundsException e) {
+                    ui.taskMissingInformationMessage();
+                } catch (DateTimeParseException e) {
+                    ui.wrongDateTimeFormatMessage();
+                } catch (FileNotFoundException e) {
+                    ui.fileNotFoundMessage();
+                }
+            }
+            break;
+        case "event":
+            if (parsedInput.isInvalidEvent()) {
+                ui.taskMissingInformationMessage();
+            } else {
+                try {
+                    String description = parsedInput.getEventDescription();
+                    LocalDateTime start = parsedInput.getStartDateTime();
+                    LocalDateTime end = parsedInput.getEndDateTime();
+                    taskList.addTask(new Event(description, false, start, end));
+                    ui.confirmTaskAddition(taskList);
+                    storage.saveFileData(taskList);
+                } catch (IndexOutOfBoundsException e) {
+                    ui.taskMissingInformationMessage();
+                } catch (DateTimeParseException e) {
+                    ui.wrongDateTimeFormatMessage();
+                } catch (FileNotFoundException e) {
+                    ui.fileNotFoundMessage();
+                }
+            }
+            break;
+        case "delete":
+            try {
+                int index = parsedInput.getIndex();
+                String taskRemoved = taskList.get(index - 1).toString();
+                taskList.deleteTask(index - 1);
+                ui.taskDeletionMessage(taskRemoved, taskList);
+                storage.saveFileData(taskList);
+            } catch (IndexOutOfBoundsException e) {
+                ui.invalidIndexMessage(taskList);
+            } catch (FileNotFoundException e) {
+                ui.fileNotFoundMessage();
+            }
+            break;
+        case "find":
+            try {
+                String keyword = parsedInput.getFindKeyword();
+                ui.tasksWithKeyword(keyword, taskList);
+            } catch (IndexOutOfBoundsException e) {
+                ui.missingKeywordMessage();
+            }
+            break;
+        default:
+            ui.invalidInputMessage();
+        }
+        return false;
```

Comment:
I think you can create your own exception and catch it at the end of the switch case statement so it will only handle your custom exception instead of handling all exceptions in every case.